### PR TITLE
feat: added banner component

### PR DESCRIPTION
**Pull request by sowmyadev-kota**
#### Description of changes:
This will add banner component to the docs website.

Staging site: https://banner.d2a6kqhihnqj53.amplifyapp.com/
### Instructions

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Comment by hibler13** [comment]
Please change the text in the banner to the following:

The next generation of backend building with Amplify!
Interested in exploring Amplify Infrastructure from Code (Amplify IfC)? Visit the documentation for our reimagined experience for web developers.

And please change the CTA text to "Go to Amplify IfC docs"

**Comment by timngyn** [comment]
@sowmyadev-kota, this component is going to be on `mdx` pages too, right? If so I think you could also add this component to this file so that content authors don't have to import the component inside the markdown: https://github.com/aws-amplify/docs/blob/next-release/main/mdx-components.tsx

**Review by hbuchel — COMMENTED** [review]
Looks good! Just two suggested changes. I think we'll need to figure out how to make it more responsive but let's get it merged in first.

File: src/styles/banner.scss [hunk 1]
```diff
@@ -0,0 +1,14 @@
+.message-banner {
+  background-color: var(--amplify-colors-purple-10) !important;
+  color: var(--amplify-colors-purple-80) !important;
```
[the remark this answers is not in the document]
Comment (sowmyadev-kota) [reply src/styles/banner.scss:3]: makes sense thank you

File: src/styles/banner.scss [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  border-radius: 12px;
+}
+.button-banner {
+  background-color: var(--amplify-colors-purple-10);
+  color: var(--amplify-colors-purple-80);
+  border-color: var(--amplify-colors-purple-80);
+  margin-top: -1.3rem;
+  margin-right: 3rem;
+  border-width: 0.06rem;
+  border-radius: 0.8rem;
```
[the remark this answers is not in the document]
Comment (sowmyadev-kota) [reply src/styles/banner.scss:13]: thank you!

File: src/styles/banner.scss (hunk 1 above)
[the remark this answers is not in the document]
Comment (sowmyadev-kota) [reply src/styles/banner.scss]: Also made changes so that banner can look good in mobile version

**Review by hbuchel — APPROVED** [review]
We need to figure out how we're going to display this and on what pages, but for now I think the component itself can be merged in. The colors will be updated via the [gen2 theme PR ](https://github.com/aws-amplify/docs/pull/6059). Thanks @sowmyadev-kota

**Review by timngyn — APPROVED** [review]

**Pull request merged**